Sources/PureSwiftJSONParsing/JSONValue.swift

```diff
@@ -131,3 +131,117 @@ public func == (lhs: JSONValue, rhs: JSONValue) -> Bool {
     return false
   }
 }
+
+@available(*, deprecated, message: "Don't use this, this is slow")
+class WritableBuffer {
+  private var ptr: UnsafeMutableRawBufferPointer
+  
+  @usableFromInline typealias _Index    = UInt32
+  @usableFromInline typealias _Capacity = UInt32
+  
+  @usableFromInline var _index   : _Index    = 0
+  @usableFromInline var _capacity: _Capacity = 6144
+  
+  init() {
+    self.ptr = UnsafeMutableRawBufferPointer.allocate(byteCount: Int(_capacity), alignment: 0)
+  }
+  
+  @inline(__always) func write(_ ptr: UnsafeRawBufferPointer) {
+    let bytesCount = ptr.count
+    let target = UnsafeMutableRawBufferPointer(rebasing: self.ptr.dropFirst(Int(_index)))
+    target.copyBytes(from: ptr)
+    self._index += _Index(bytesCount)
+  }
+  
+  @inline(__always) func write(byte: UInt8) {
+    self.ptr[Int(self._index)] = byte
+    self._index += 1
+  }
+  
+  @inline(__always) func toBytes() -> [UInt8] {
+    return [UInt8](ptr)
+  }
+}
+
+extension WritableBuffer {
+  
+  func write(json: JSONValue) {
+    
+    switch json {
+    case .null:
+      self.write(byte: UInt8(ascii: "n"))
+      self.write(byte: UInt8(ascii: "u"))
+      self.write(byte: UInt8(ascii: "l"))
+      self.write(byte: UInt8(ascii: "l"))
+      
+    case .bool(true):
+      self.write(byte: UInt8(ascii: "t"))
+      self.write(byte: UInt8(ascii: "r"))
+      self.write(byte: UInt8(ascii: "u"))
+      self.write(byte: UInt8(ascii: "e"))
+      
+    case .bool(false):
+      self.write(byte: UInt8(ascii: "f"))
+      self.write(byte: UInt8(ascii: "a"))
+      self.write(byte: UInt8(ascii: "l"))
+      self.write(byte: UInt8(ascii: "s"))
+      self.write(byte: UInt8(ascii: "e"))
+      
+    case .string(let string):
+      self.write(byte: UInt8(ascii: "\""))
+      string.utf8.withContiguousStorageIfAvailable {
+        self.write(UnsafeRawBufferPointer($0))
+      }
+      self.write(byte: UInt8(ascii: "\""))
+    case .number(let string):
+      string.utf8.withContiguousStorageIfAvailable {
+        self.write(UnsafeRawBufferPointer($0))
+      }
+    case .array(let array):
+      var iterator = array.makeIterator()
+      self.write(byte: UInt8(ascii: "["))
```

Comment:
Iterators copy values more often than needed

Reply:
Am I correct in assuming that you would vote for direct array access? for elem in array?